### ISSUE TITLE
12hr format for Timepicker not correct.

### ISSUE THREAD

**Issue by gabrielholley** [uikit/uikit#1113]
data-uk-timepicker="{format:'12h'}"

Is not correct.

The list of options should be:

12:00 AM
12:30 AM
01:00 AM
01:30 AM
02:00 AM
02:30 AM
03:00 AM
03:30 AM
04:00 AM
04:30 AM
05:00 AM
05:30 AM
06:00 AM
07:00 AM
07:30 AM
08:00 AM
08:30 AM
09:00 AM
09:30 AM
10:00 AM
10:30 AM
11:00 AM
11:30 AM
12:00 PM
12:30 PM
01:00 PM
01:30 PM
02:00 PM
02:30 PM
03:00 PM
03:30 PM
04:00 PM
04:30 PM
05:00 PM
05:30 PM
06:00 PM
07:00 PM
07:30 PM
08:00 PM
08:30 PM
09:00 PM
09:30 PM
10:00 PM
10:30 PM
11:00 PM
11:30 PM

I've attached images of what it is now.

![screen shot 2015-03-28 at 12 34 28 am](https://cloud.githubusercontent.com/assets/4369156/6880080/3975996e-d4e3-11e4-8ee8-6351240b9a92.png)
![screen shot 2015-03-28 at 12 34 57 am](https://cloud.githubusercontent.com/assets/4369156/6880081/39788ab6-d4e3-11e4-8818-3cc1e75603a6.png)

**Comment by malles** [comment]
I think this will be fixed by #1066, submitted a few weeks ago.

**Comment by gabrielholley** [comment]
#1066 Doesn't fix. The whole loop for 12 hour needs to be re-thought. Each 12 in that format is really zero and needs to be first in each meridian time block . The list ordering in my first comment is the correct order.

Attached image of #1066 in use. 12 AM and 12:30 AM are out of order.

![screen shot 2015-03-31 at 9 19 34 pm](https://cloud.githubusercontent.com/assets/4369156/6934075/48543182-d7ec-11e4-8816-485ef6b07bc9.png)

This is correct loop for timepicker.js:

```
    var times = {'12h':[], '24h':[]};

    for(var i = 0, h=''; i<24; i++) {

        h = ''+i;

        if(i<10)  h = '0'+h;

        times['24h'].push({value: (h+':00')});
        times['24h'].push({value: (h+':30')});

        if(i == 0) {
            h = 12;
            times['12h'].push({value: (h+':00 AM')});
            times['12h'].push({value: (h+':30 AM')});
        }

        if (i > 0 && i<13 && i!==12) {
            times['12h'].push({value: (h+':00 AM')});
            times['12h'].push({value: (h+':30 AM')});
        }

        if (i >= 12) {

            h = h-12;
            if (h == 0) h = 12;
            if (h < 10) h = '0'+String(h);

            times['12h'].push({value: (h+':00 PM')});
            times['12h'].push({value: (h+':30 PM')});
        }
    }
```

![screen shot 2015-03-31 at 9 41 36 pm](https://cloud.githubusercontent.com/assets/4369156/6934265/e5ae7ecc-d7ee-11e4-81d0-9cbdb9f9a366.png)
![screen shot 2015-03-31 at 9 42 04 pm](https://cloud.githubusercontent.com/assets/4369156/6934266/e5b0d30c-d7ee-11e4-96ed-6009143aba97.png)
